zerver/views/registration.py

```diff
@@ -137,7 +137,7 @@ def accounts_register(request):
             form = RegistrationForm(
                 initial={'full_name': hesiod_name if "@" not in hesiod_name else ""})
             name_validated = True
-        elif settings.POPULATE_PROFILE_VIA_LDAP:
+        elif settings.POPULATE_PROFILE_VIA_LDAP and not realm_creation:
```

Comment:
I think the invalid form that we get due ToS field in the first case is also a logical bug even though we get the correct result.